ctest-next/tests/sources/macro.rs

```diff
@@ -0,0 +1,13 @@
+macro_rules! t {
+    ($e:expr) => {
+        match $e {
+            Ok(e) => e,
+            Err(e) => panic!("{} failed with {}", stringify!($e), e),
+        }
+    };
+}
+
+pub fn write_buffer() {
+    let buf = String::new();
+    t!(writeln!(buf, "#define A 1"));
+}
```

Comment:
Could the macro generate structs or functions that will be visible in the C interface? Since that is what we'll want to test.